src/db.rs

```diff
@@ -506,7 +519,25 @@ mod tests {
         db.insert_secret(&secid_uuid, &sec_uuid, Option::Some(tpid))
             .await?;
 
-        let testpol = db.get_secret_policy(&secid_uuid).await?;
+        let testpol = match db.get_secret_policy(&secid_uuid).await {
+            Ok(tpopt) => match tpopt {
+                Some(tp) => tp,
+                None => policy::Policy {
+                    allowed_digests: vec!["".to_string()],
+                    allowed_policies: vec![],
+                    min_fw_api_major: 0,
+                    min_fw_api_minor: 0,
+                    allowed_build_ids: vec![],
+                },
+            },
+            Err(_e) => policy::Policy {
+                allowed_digests: vec!["".to_string()],
+                allowed_policies: vec![],
+                min_fw_api_major: 0,
+                min_fw_api_minor: 0,
+                allowed_build_ids: vec![],
+            },
+        };
```

Comment:
I don't quite understand this match statement. Why don't we just unwrap the policy?

Reply:
On top of that: do we have a test that verifies that our code works with NULL policy? That test should fail without the suggested fixes to `get_secret_policy()` and `get_keyset_policy()` above.